components/Labels/ManageLabelBox.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect, useState } from "react";
+import { Button, Chip, Icon, Scrim, Search } from "@equinor/eds-core-react";
+import styles from "./ManageLabelBox.module.scss";
+import { close } from "@equinor/eds-icons";
+import {
+  addLabelToProcess,
+  getLabels,
+  removeLabelFromProcess,
+} from "services/labelsApi";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { vsmProject } from "interfaces/VsmProject";
+import { processLabel } from "interfaces/processLabel";
+import { debounce } from "utils/debounce";
+import { unknownErrorToString } from "utils/isError";
+
+export default function ManageLabelBox(props: {
+  isVisible: boolean;
+  handleClose: () => void;
+  process: vsmProject;
+}): JSX.Element {
+  if (!props.isVisible) return null;
+
+  return (
+    <Scrim onClose={props.handleClose} isDismissable>
+      <div className={styles.box} onWheel={(e) => e.stopPropagation()}>
+        <TopSection handleClose={props.handleClose} />
+        <AddSection process={props.process} />
+      </div>
+    </Scrim>
+  );
+}
+
+function TopSection(props: { handleClose: () => void }): JSX.Element {
+  return (
+    <div className={styles.topSection}>
+      <p className={styles.heading}>Add labels</p>
+      <Button variant={"ghost_icon"} onClick={props.handleClose}>
+        <Icon data={close} />
+      </Button>
+    </div>
+  );
+}
+
+function AddSection(props: { process: vsmProject }): JSX.Element {
+  const [term, setTerm] = useState("");
+  const [debouncedTerm, setDebouncedTerm] = useState("");
+  const queryClient = useQueryClient();
+
+  const { data: labels, error } = useQuery(["labels", debouncedTerm], () =>
+    getLabels(debouncedTerm)
+  );
+
+  const handleChange = (event) => {
+    setTerm(event.target.value);
+  };
+
+  useEffect(() => {
+    debounce(
+      () => {
+        setDebouncedTerm(term);
+      },
+      200,
+      "LabelSearchQuery"
+    );
+  }, [term]);
+
+  const addLabelMutation = useMutation(
+    (payload: {
+      processID: number;
+      label: processLabel | { text: string };
+    }) => {
+      return addLabelToProcess(payload.processID, payload.label);
+    },
+    { onSettled: () => queryClient.invalidateQueries() }
+  );
+
+  const handleSelect = (item: string) => {
+    const trimmedItem = item.trim();
+    if (!trimmedItem) {
+      return null;
+    }
+    addLabelMutation.mutate({
+      processID: props.process.vsmProjectID,
+      label: { text: trimmedItem },
+    });
+    props.process.toBeProcessID &&
+      addLabelMutation.mutate({
+        processID: props.process.toBeProcessID,
+        label: { text: trimmedItem },
+      });
+    props.process.currentProcessId &&
+      addLabelMutation.mutate({
+        processID: props.process.currentProcessId,
+        label: { text: trimmedItem },
+      });
```

Comment:
@Magos Correct me if I'm mistaken… I thought we were going to handle this in the API? Ref https://github.com/equinor/MAD-VSM-WEB/issues/159#issuecomment-998734537

Reply:
Status: I have removed this piece of code as it should actually be handled by the API. There was a bug (when adding a label to the to-be process, the current one is not updated) that @Magos is currently fixing.

Reply:
Great! Awaiting code changes